diff --git a/dist/vitest-plugin/test-utils.js b/dist/vitest-plugin/test-utils.js
index 9048740b2c8fba5a813a5541180dbd8bc2ec5984..58d0b778b93f472b5cef04edb174c8f104c27616 100644
--- a/dist/vitest-plugin/test-utils.js
+++ b/dist/vitest-plugin/test-utils.js
@@ -51,10 +51,10 @@ var DEFAULT_VIEWPORT_DIMENSIONS = {
 }, setViewport = async (parameters = {}, globals = {}) => {
   let defaultViewport, viewportsParam = parameters.viewport ?? {}, viewportsGlobal = globals.viewport ?? {}, isDisabled = viewportsParam.disable || viewportsParam.disabled;
   viewportsGlobal.value && !isDisabled ? defaultViewport = viewportsGlobal.value : isDisabled || (defaultViewport = viewportsParam.defaultViewport);
-  let { page } = await import("@vitest/browser/context").catch(() => ({
+  let { page } = await import("vitest/browser").catch(() => ({
     page: null
   }));
-  if (!page || !globalThis.__vitest_browser__)
+  if (!page)
     return;
   let options = {
     ...MINIMAL_VIEWPORTS,
